padertorch/ops/losses/loss.py

```diff
@@ -105,16 +152,24 @@ def pit_loss(estimate, target, loss_fn=torch.nn.functional.mse_loss):
             f'{estimate.size()} != {target.size()}'
         )
     candidates = []
-    for permutation in itertools.permutations(range(sources)):
+    filler = (slice(None),) * axis
+    permutations = list(itertools.permutations(range(sources)))
+    for permutation in permutations:
         candidates.append(loss_fn(
-            estimate[:, permutation, :],
+            estimate[filler + (permutation, )],
             target
         ))
-    return torch.min(torch.stack(candidates))
+
+    min_loss, idx = torch.min(torch.stack(candidates), dim=0)
+
+    if return_permutation:
+        return min_loss, permutations[int(idx)]
+    else:
+        return min_loss
 
 # this function is kept at the moment for backwards compatibility
 # ToDo: remove this function
-pit_mse_loss = pit_loss
+pit_mse_loss = partial(pit_loss, axis=-2)
```

Comment:
I would suggest removing lines 169 to 172 and keep mse as default loss for pit

Reply:
`pit_mse_loss` is used in the `PermutationInvariantTraining` models. If we remove lines 169 to 172 we have to touch those models as well

Reply:
And some other people might use it in their personal repositories

Reply:
In that case I am fine with keeping it for the moment and removing it in a separate pull request together with all uses in padertorch.